pkg/common/http.go

```diff
@@ -66,7 +66,7 @@ func HTTPResponseBody(httpResponse *http.Response) ([]byte, error) {
 	}
 
 	if httpResponse.Status != "200 OK" && httpResponse.Status != "200 " {
-		err = errors.Errorf("http response status equal to %s", httpResponse.Status)
+		err = errors.Wrapf(errors.Errorf("http response status equal to %s", httpResponse.Status), string(httpResponseBodyBytes))

```

Comment:
You would also likely want to include the bytes in the other return above.  It's possible to get a partial read before the connection is unexpectedly closed which could provide insight to the issue.

Reply:
I ported the changes you made in demos repo to here so that the partial read before connection close won't be an issue.